Add wheel slip message definition

ivanpauno opened the pull request:
## Summary
<!--Explain changes made, the expected behavior, and provide any other additional
context (e.g., screenshots, gifs) if appropriate.-->
Adds a build sleep message definition, which would be used to reconfigure the wheel slip systems parameters.
The slip compliance parameters used here are unitless, as explained in the [wheel slip system](https://github.com/ignitionrobotics/ign-gazebo/blob/a06198fb95001078dc45492a52458252f31d86bc/src/systems/wheel_slip/WheelSlip.hh#L43-L44).
Maybe we should be more explicit about that in the msg definition (?).

## Test it
<!--Explain how reviewers can test this new feature manually.-->

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

ivanpauno commented:
@scpeters I'm targetting ign-msgs7 because I'm using `edifice` locally for development, but maybe that was an incorrect choice.
What version of ignition should I be targetting?

scpeters commented:
> @scpeters I'm targetting ign-msgs7 because I'm using `edifice` locally for development, but maybe that was an incorrect choice.
> What version of ignition should I be targetting?

we generally try to target the oldest version that will support the feature, but I'm not sure which version that will be yet. I think Edifice is fine for now, though it will EOL in March 2022

scpeters commented:
one thing to keep in mind is that we have just added a [WheelSlip.msg](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_msgs/msg/WheelSlip.msg) to `gazebo_msgs` in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1331 for publishing instantaneous wheel slip, so we may want to rename the message proposed here to `wheel_slip_parameters` or `wheel_slip_configuration` to avoid confusion

ivanpauno commented:
> one thing to keep in mind is that we have just added a WheelSlip.msg to gazebo_msgs in ros-simulation/gazebo_ros_pkgs#1331 for publishing instantaneous wheel slip, so we may want to rename the message proposed here to wheel_slip_parameters or wheel_slip_configuration to avoid confusion

Addressed in https://github.com/ignitionrobotics/ign-msgs/pull/205/commits/a44f73eaf422891823cad51bca272d1b22f5337b

scpeters commented on proto/ignition/msgs/wheel_slip_parameters.proto:
```diff
@@ -0,0 +1,46 @@
+  string name         = 2;
+  uint32 id           = 3;
+  string parent_name  = 4;
+  uint32 parent_id    = 5;
```
wait, I think these fields are used...

scpeters replied:
related to https://github.com/ignitionrobotics/ign-gazebo/pull/1241/files#r782616278

ivanpauno replied:
> related to https://github.com/ignitionrobotics/ign-gazebo/pull/1241/files#r782616278

That comment was resolved in ign-gazebo#1241

ivanpauno commented on proto/ignition/msgs/wheel_slip_parameters.proto:
```diff
@@ -0,0 +1,46 @@
+import "ignition/msgs/geometry.proto";
+import "ignition/msgs/material.proto";
+import "ignition/msgs/plugin.proto";
+import "ignition/msgs/vector3d.proto";
```
Fixed in https://github.com/ignitionrobotics/ign-msgs/pull/205/commits/f154c103796271a60ad2329802170868e3fc4d60

ivanpauno commented on proto/ignition/msgs/wheel_slip_parameters.proto:
```diff
@@ -0,0 +1,46 @@
+
+/// \ingroup ignition.msgs
+/// \interface Visual
+/// \brief A message containing visual information for rendering::Visual
```
Done in https://github.com/ignitionrobotics/ign-msgs/pull/205/commits/f154c103796271a60ad2329802170868e3fc4d60.

ivanpauno commented on proto/ignition/msgs/wheel_slip_parameters.proto:
```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "VisualProtos";
```
Fixed, see https://github.com/ignitionrobotics/ign-msgs/pull/205/commits/f154c103796271a60ad2329802170868e3fc4d60.

ivanpauno commented:
When working in a demo of this using parameters, I realized in that case I need a slightly different message definition https://github.com/ignitionrobotics/ign-msgs/commit/a5b39fdebffe55ea442b37fa1e638400e64b6854.

Maybe we should rename the one being added here `WheelSlipParametersCmd`, what do you think?

jacobperron reviewed (approved):
LGTM

chapulina commented on proto/ignition/msgs/wheel_slip_parameters_cmd.proto:
```diff
@@ -0,0 +1,39 @@
+  string link_name                    = 3;
+
+  double slip_compliance_lateral      = 4;
+  double slip_compliance_longitudinal = 5;
```
Sorry for the late comment, but while reviewing the forward-port, I noticed that most fields on this message aren't documented.

ivanpauno replied:
I can add docs in a new PR

ivanpauno replied:
See https://github.com/ignitionrobotics/ign-msgs/pull/227

osrf-triage commented:
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-25-fortress-edifice-citadel/1343/1
